gms/versioned_value.hh

```diff
@@ -209,3 +209,9 @@ public:
 }; // class versioned_value
 
 } // namespace gms
+
```

Comment:
> in this change, we define formatters for `gms::versioned_value`. its
operator<< is preserved, as it's still being used by the homebrew
generic formatter for std::unordered_map<gms::application_state,
gms::versioned_value>, which is in turn used in gms/gossiper.cc.


In such cases it's better to rewrite operator<< in terms of the formatter, so we have only format, and so the reviewer can easily check against the original.

Reply:
sure. created #17390 to address this comment.